src/stream.rs

```diff
@@ -11,27 +11,29 @@ use chunked_transfer::Decoder as ChunkDecoder;
 #[cfg(feature = "socks-proxy")]
 use socks::{TargetAddr, ToTargetAddr};
 
+#[cfg(not(feature = "native-tls"))]
+use crate::default_tls_config;
 use crate::proxy::Proxy;
 use crate::{error::Error, proxy::Proto};
 
 use crate::error::ErrorKind;
 use crate::unit::Unit;
 
 /// Trait for things implementing [std::io::Read] + [std::io::Write]. Used in [TlsConnector].
-pub trait ReadWrite: Read + Write + Send + 'static {
+pub trait ReadWrite: Read + Write + Send + Sync + 'static {
     fn socket(&self) -> Option<&TcpStream>;
 }
 
 pub trait TlsConnector: Send + Sync {
     fn connect(
         &self,
         dns_name: &str,
-        tcp_stream: TcpStream,
+        tcp_stream: Stream,
```

Comment:
This is also a breaking change in our public API.

Reply:
Yes, for HTTPS proxy support we require a TlsConnector (destination) wrapping a TlsConnector (proxy).
I couldn't think of a different solution that would be less intrusive and wouldn't require a public API change.

As you're more familiar with the code, does anything come to mind on how to solve this differently?